qiskit_nature/results/utils/protein_plotter.py

```diff
@@ -0,0 +1,144 @@
+# that they have been altered from the originals.
+"""An auxiliary class that plots aminoacids of a molecule
+ in a ProteinFoldingResult."""
+import matplotlib.pyplot as plt
+import numpy as np
+import qiskit_nature.optionals as _optionals
+
+from qiskit_nature.results.utils.protein_shape_file_gen import ProteinShapeFileGen
+
+
+@_optionals.HAS_MATPLOTLIB.require_in_instance
+class ProteinPlotter:
+    """Plotter class for ProteinFoldingResult."""
+
+    def __init__(self, shape_gen: ProteinShapeFileGen) -> None:
+        """
+        Args:
+            protein_folding_result: The protein folding result to be plotted
+        """
+
+        self._shape_gen = shape_gen
+        # pylint: disable=unbalanced-tuple-unpacking
+        (
+            self._x_main,
+            self._y_main,
+            self._z_main,
+        ) = np.split(self._shape_gen.main_positions.transpose(), 3, 0)
+        self._x_main, self._y_main, self._z_main = self._x_main[0], self._y_main[0], self._z_main[0]
+
+        self._fig = plt.figure()
+        self._ax_graph = self._fig.add_subplot(projection="3d")
+
+    def _draw_main_chain(self):
+        """
+        Draws the main chain.
+
+        """
+        for i, main_aminoacid in enumerate(self._shape_gen._main_chain_aminoacid_list):
+            self._ax_graph.text(
+                self._x_main[i],
+                self._y_main[i],
+                self._z_main[i],
+                main_aminoacid,
+                size=10,
+                zorder=10,
+                color="k",
+            )
+
+        self._ax_graph.plot3D(self._x_main, self._y_main, self._z_main)
+        return self._ax_graph.scatter3D(
+            self._x_main, self._y_main, self._z_main, s=500, label="Main Chain"
+        )
+
+    def _draw_side_chains(self):
+        """
+        Draws the side chain.
+        """
+        side_positions = self._shape_gen.side_positions
+        side_aminoacids = self._shape_gen._main_chain_aminoacid_list
+        for i, side_chain in enumerate(side_positions):
+            if side_chain is not None:
+                x_side, y_side, z_side = side_chain
+                side_scatter = self._ax_graph.scatter3D(
+                    x_side, y_side, z_side, s=600, c="green", label="Side Chain"
+                )
+                self._ax_graph.plot3D(
+                    [self._x_main[i], x_side],
+                    [self._y_main[i], y_side],
+                    [self._z_main[i], z_side],
+                    c="green",
+                )
+                self._ax_graph.text(
+                    x_side,
+                    y_side,
+                    z_side,
+                    side_aminoacids[i],
+                    size=10,
+                    zorder=10,
+                    color="k",
+                )
+        return side_scatter
+
+    def _format_graph(
+        self, title: str, ticks: bool, grid: bool, main_scatter: plt.Axes, side_scatter: plt.Axes
```

Comment:
Again, I suggest to allow customization via `kwargs` rather than a chosen subset of options